Automatic page routing & port handling

chrimesdev opened the pull request:
Automatic page routing
Port handling
Remove default passwords

stevenadams commented:
Hi Paul,
I'd like to keep this work in the mvp branch for now as I don't have time to check it fully. I will merge styles into this branch before I leave.

paul-mitchell-bjss commented:
That's totally fine. It doesn't need to be on master until we're all happy with it.

chrimesdev committed:
Handle the common port in use error with express

If the selected port is being used by another process then this
app will crash. Instead of crashing the app now attempts on run
on a randomised port number.

chrimesdev committed:
Fix incorrect nhsuk.min.js asset path from nhsuk-frontend

chrimesdev committed:
Add automatic page routing and rendering

Try to match a request to a template, for example a request for /test
would look for /app/views/test.html and /app/views/test/index.html

chrimesdev committed:
Prioritise the env port over the config.js port

Set the default port to 3000 in the config port but prioritise
the env port over it.

chrimesdev committed:
Remove default password and password setting from config.js

Make it so the passwords have to be set on within the environment
variables when running the application in production.
